portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -13,27 +16,52 @@
 /** Servlet that handles user authentication, so that users must log in to post comments */
 @WebServlet("/login") 
 public class AuthenticationServlet extends HttpServlet {
-  //TODO: change to actual redirect page. currently just redirects to test logoff
-  private final static String REDIRECT_URL = "/login.html"; 
-  private static final String JSON_CONTENT_TYPE = "application/json;";
+  private final static String DISPLAY_NAME_URL = "/displayname.html";
+  // TODO: change to actual logoff redirection page
+  private final static String REDIRECT_LOGOFF_URL = "/login.html";
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  UserService userService; 
+  UserDatabase userDatabase;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    UserService userService = UserServiceFactory.getUserService();
+    this.userService = UserServiceFactory.getUserService();
     Boolean isUserLoggedIn = userService.isUserLoggedIn();
-    String userEmail;
-    String url;
+    UserAuthentication userAuthentication;
+
     if (isUserLoggedIn) {
-        userEmail = userService.getCurrentUser().getEmail();
-        url = userService.createLogoutURL(REDIRECT_URL);
+        userAuthentication = processLoggedInUser(userService);
     } else {
-        userEmail = "";
-        url = userService.createLoginURL(REDIRECT_URL);
+        userAuthentication = processNotLoggedInUser(userService);
     }
-    UserAuthentication userAuthentication = new UserAuthentication(isUserLoggedIn, userEmail, url);
+
     Gson gson = new Gson();
     String userAuthenticationInfoJSON = gson.toJson(userAuthentication); 
     response.setContentType(JSON_CONTENT_TYPE);
     response.getWriter().println(userAuthenticationInfoJSON);
   }
+
+  private UserAuthentication processNotLoggedInUser(UserService userService) {
+    Boolean isUserLoggedIn = false;
+    String userEmail = "";
```

Comment:
Are these always empty? Why assign them to variables? We could just pass String.empty into the UserAuthentication constructor, right?